### ISSUE TITLE
Add CommonJS support

### ISSUE THREAD

**Issue by j0nas** [fenok/react-router-typesafe-routes#30]
I'm using react-router-typesafe-routes in a TypeScript project that compiles with Vite. Vite has no issues with the ESM-only output of this package, but I am unable to run my Playwright tests because they require CommonJS imports.

```
Error [ERR_REQUIRE_ESM]: require() of ES Module /Users/user/Desktop/path/to/project/node_modules/react-router-typesafe-routes/dom/index.js from /Users/user/Desktop/ruter/path/to/project/src/routes.ts not supported.
Instead change the require of index.js in /Users/jonasjensen/Desktop/ruter/bestillingstransport/apps/bt-frontend/src/admin/routing/routes.ts to a dynamic import() which is available in all CommonJS modules.

   at ../src/admin/routing/routes.ts:7

   5 |   OVERVIEW: route('overview'),
   6 |   PROFILE: route('profile'),
>  7 |   TRIPS: route('trips', {}, {LIVE: route('live')}),
     |            ^
   8 |   MORE_ROUTES: route('more_routes', {}, {CLIENTS: route('clients')}),
```

 I unfortunately discovered this only _after_ I'd already rewrite all hardcoded paths in my entire codebase to use `react-router-typesafe-routes`. 😬  It would be very helpful if the package could support both ESM and CommonJS formats. 

It would be great if `react-router-typesafe-routes` could provide a CommonJS output alongside the ESM output. This would make the package compatible with various testing environments and setups, including those that rely on CommonJS.

(As a workaround, I've tried bundling and transforming my tests using esbuild, but this adds too much extra complexity and I couldn't even get it to work properly..)

Thank you for considering this feature request, and I appreciate your efforts in creating and maintaining this package! I compared all the different "typesafe routes" packages that I could find and this one was by far the best I could find. 🙌

**Comment by fenok** [comment]
Thank you! Unfortunately, I don't have experience in creating packages that provide both ESM and CommonJS. It seems pretty easy to mess up.

I'll try to implement something like [this](https://www.sensedeep.com/blog/posts/2021/how-to-create-single-source-npm-module.html), but it will take some time, and I expect additional issues from the fact that the library has multiple entry points.

I also want to roll out https://github.com/fenok/react-router-typesafe-routes/pull/29 first.

If you know a better approach to hybrid packages, please share. The one above feels ugly 😅.

A PR is also welcome.

**Comment by fenok** [comment]
I tried to use Microbundle some time ago, but I had some issues and resorted to using Rollup directly. I'll give it a try though, thanks for reminding me about it.

> do you happen to have an ETA for that?

Somewhere within a week, if everything goes smoothly.

**Comment by j0nas** [comment]
Awesome. Let me know how it goes, I definitely wouldn't be opposed to picthing in if need be. :)

**Comment by fenok** [comment]
In the end, I went with Microbundle. I published CommonJS support under the `next` tag. It seems to work at the first glance, but I want to test it a bit more.

**Comment by fenok** [comment]
Wait, I realized that microbundle produces independent bundles, which leads to code duplication. What I actually need is bundles that depend on each other. For instance, `zod.js` should import stuff from `index.js` (and `zod.cjs` should require stuff from `index.cjs`). Maybe bundling is a bad idea altogether.

**Comment by j0nas** [comment]
That doesn't sound right, why would you want that? The three bundles should indeed be separate. 

I'll get to testing this ASAP and will let you know.

**Comment by j0nas** [comment]
@fenok `1.2.0-dev.4` works great, fantastic job! 

`1.1.0`: 
![image](https://user-images.githubusercontent.com/6001131/233627916-265ff964-3b01-48bc-9920-340947a8c449.png)


`1.2.0-dev.4`:
![image](https://user-images.githubusercontent.com/6001131/233628348-70cf5ced-4bfa-4dc3-8859-3a38ab6170d4.png)

**Comment by j0nas** [comment]
![image](https://user-images.githubusercontent.com/6001131/233722925-a5688428-8efa-4b0d-8fe4-6a7a1524ccff.png)

`dev.5` works equally well 🙌 

Alright, thanks for the recap. I don't have much experience with bundling for package. Very grateful that you took the time and effort to look into this! ❤️

**Comment by fenok** [comment]
Released in `v1.2.0`.
